### PR TITLE
Helps CC-1248 Panic when no network configured

### PR DESCRIPTION
**Pull request by kwalkerz**
There were several panics in the cli code, and some were early enough to
prevent simple subcommands (like 'help' and 'version') from running. I
changed them to log instead (often Fatal, so serviced would shut down)
so we don't have the stack traces, which are bothersome to customers
(and they didn't seem like they would be helpful for the problems that
were being detected anyway).

I also found that starting the server ignored some errors that were
returned--we now log those as Fatal errors (so we shut down).

### CODE REVIEW

File: cli/cmd/options.go [hunk 1]
```diff
@@ -100,7 +100,8 @@ func getDefaultEndpoint(ip, port string) string {
 	if ip == "" {
 		var err error
 		if ip, err = utils.GetIPAddress(); err != nil {
-			panic(err)
+			//fmt.Fprintf(os.Stderr, "ERROR: Unable to get default endpoint: %s\n", err)
+			return ""
```
Comment (smousa) [inline cli/cmd/options.go]: Should you fatal here as well?

File: cli/cmd/utils.go [hunk 2]
```diff
@@ -35,7 +37,7 @@ func remove(index int, list ...interface{}) []interface{} {
 
 	switch {
 	case index < 0 || index > len(list):
-		panic("index out of bounds")
```
Comment (smousa) [inline cli/cmd/utils.go]: This function isn't called anywhere, can you just delete this?

File: cli/cmd/options.go (hunk 1 above)
Comment (kwalkerz) [inline cli/cmd/options.go]: I did not use Fatal here so simple commands like 'serviced version' and 'serviced help' would run.  This "" gets placed in options.Endpoint, then when the server actually tries to start up and discovers it doesn't have the info it needs, it Fatals out.  (Even printing the error message that I commented out is very distracting and annoying for those simple commands.)

Comment (smousa) [inline cli/cmd/options.go]: ok, make sure you clean up your dead code then

File: cli/cmd/utils.go (hunk 2 above)
Comment (kwalkerz) [inline cli/cmd/utils.go]: Even better--will do.

**Comment by smousa** [comment]
waiting for tests to pass

**Pull request merged**